pipe_segment/transform/create_segment_map.py

```diff
@@ -51,28 +90,35 @@ def compute_scores(self, segs, frag_ids, frag_map):
                 scores[sid, fid] = self.compute_pair_score(frag0, frag1)
         return scores
 
-    def frags_by_day(self, frags):
+    def frags_by_day(
+        self, frags: Iterable[dict]
+    ) -> Generator[Tuple[date, Set[dict]], None, None]:
         for day, frags_by_day in by_day(frags):
-            yield day, [x["frag_id"] for x in frags_by_day]
+            yield day, {x["frag_id"] for x in frags_by_day}
```

Comment:
@bitsofbits should it be a `list` instead of a `dict`? because it will collect the frag_ids of the day..

Reply:
@smpiano : is is actually a set, not a dict here.  We don't care about the order, and this prevents duplicates (which I don't think we care about here) and more importantly allows items to be removed in constant time (as opposed to a list, which is linear time I believe). And we remove items from this down on line 121.